src/main/java/core/basesyntax/dao/StorageDaoCsvImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitCrate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class StorageDaoCsvImpl implements StorageDao {
+    @Override
+    public FruitCrate update(FruitCrate fruitCrate) {
+        if (get(fruitCrate.getName()) == null) {
+            Storage.storage.add(fruitCrate);
+        } else {
+            int newQuantity = fruitCrate.getQuantity() + get(fruitCrate.getName()).getQuantity();
+            get(fruitCrate.getName()).setQuantity(newQuantity);
+        }
```

Comment:
let's move all this logic out of this method to ActivityHandlers, the only line that must remain is 12th for the case if FruitCrate is new